partner_relations/model/res_partner.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+'''Extend res.partner model'''
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2013 Therp BV (<http://therp.nl>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import time
+from openerp import osv, models, fields, exceptions, api
+from openerp.osv.expression import is_leaf, AND, OR, FALSE_LEAF
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+from openerp.tools.translate import _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    relation_count = fields.Integer(
+        'Relation Count',
+        compute="_count_relations"
+    )
+
+    @api.one
+    @api.depends("relation_ids")
+    def _count_relations(self):
+        """Count the number of relations this partner has for Smart Button
+
+        Don't count inactive relations.
+        """
+        self.relation_count = len([r for r in self.relation_ids if r.active])
+
+    def _get_relation_ids_select(self, cr, uid, ids, field_name, arg,
+                                 context=None):
+        '''return the partners' relations as tuple
+        (id, left_partner_id, right_partner_id)'''
+        cr.execute(
```

Comment:
there was a time when it was relevant if we found the relation as a left partner or a right partner. Meanwhile, the magic for this is done in `res.partner.relation`, so probably this can be done in the ORM by now. There might be weirdnesses casued by https://github.com/odoo/odoo/pull/5807 though, as both `left_partner_id` and `right_partner_id` are auto_joined.